numerics/polynomial.hpp

```diff
@@ -31,12 +31,12 @@ FORWARD_DECLARE_FUNCTION_FROM(
     mathematica,
     TEMPLATE(typename Value, typename Argument, int degree_,
              template<typename, typename, int> typename Evaluator,
-             typename OptionalExpressIn = std::nullopt_t) std::string,
+             typename OptionalExpressIn) std::string,
```

Comment:
This is fishy.  Similar declarations exist in `poisson_series.hpp` and `piecewise_poisson_series.hpp`, so I presume that they would need changing too.  As a matter of fact, it's seems better to have the default live in `mathematica` than in distant units that happen to have a forward declaration.

Reply:
> Similar declarations exist in `poisson_series.hpp` and `piecewise_poisson_series.hpp`

They don’t (they don’t have defaults).

> As a matter of fact, it's seems better to have the default live in `mathematica` than in distant units that happen to have a forward declaration.

This is what this pull request does.